### DISCUSSION TITLE
Improve `print.tinyVAST`

### DISCUSSION THREAD

**Discussion by James-Thorson-NOAA** [vast-lib/tinyVAST#41] [Ideas]
@seananderson Do you mind if I just copy over your code for `print.sdmTMB` (with small modification) to improve `print.tinyVAST`?  And any thoughts or suggestions to share before I do so?

**Comment by seananderson** [comment]
You're very welcome to take what you need.

That function is one that I did refactor at one point, which was a good decision. It had got out of hand. Some of the lowest level functions (e.g. `print_smooth_effects()`) are a bit of a mess and just get the job done, but I'm fairly happy with the overall approach of
- making a bunch of small internal printing functions for the various parts of the model; they all take a common set of arguments (easy to write, easy to test); if that part doesn't exist, it returns NULL
- making a function that calls all those pieces for a given delta model linear predictor
- cycling over that function for each delta model part as needed

**Comment by James-Thorson-NOAA** [comment]
Thanks for the feedback!  I took a first pass, and will close for now, but plan to update this thread when I cycle back to the topic :)
